src/foundation/components/Label/index.tsx

```diff
@@ -0,0 +1,73 @@
+import { cx } from '@linaria/atomic';
+import React, { ElementType, HTMLAttributes } from 'react';
+
+import { labelStyles } from './css';
+import { textColors, themedColorVars } from '../../colors/css';
+import { useLocalProps } from '../../hooks/useLocalProps';
+import { SBUFoundationProps } from '../../types';
+
+type AsProp<C extends ElementType> = {
+  as?: C;
+};
+type PropsToOmit<C extends ElementType, P> = keyof (P & AsProp<C>);
+type PolymorphicComponentProps<C extends ElementType, Props = NonNullable<unknown>> = Props &
+  AsProp<C> &
+  Omit<React.ComponentPropsWithoutRef<C>, PropsToOmit<C, Props>>;
+
+type HTMLProps<C extends ElementType> = PolymorphicComponentProps<C, HTMLAttributes<HTMLElement>>;
+
+type Props<C extends ElementType> = SBUFoundationProps<
+  HTMLProps<C> & {
+    type?:
+      | 'h1'
+      | 'h2'
+      | 'subtitle1'
+      | 'subtitle2'
+      | 'body1'
+      | 'body2'
+      | 'button1'
+      | 'button2'
+      | 'button3'
+      | 'caption1'
+      | 'caption2'
+      | 'caption3';
```

Comment:
will these be resused? maybe define `ElementType`?

Reply:
I didn't quite understand what you are asking.